submissions/aymkin/OOP-JS (tjsw)/index.js

```diff
@@ -0,0 +1,80 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+
+Code repository: https://github.com/aymkin/a-tiny-JS-world/tree/tiny-js-world-oop-aymkin
+Web app: https://aymkin.github.io/a-tiny-JS-world/
+   */
+
+function Inhobitant(species, name, gender, friends, say) {
+	this.species = species;
+	this.name = name;
+	this.gender = gender;
+	this.friends = friends;
+	this.say = say;
+}
+
+Inhobitant.prototype.makeSound = function() {
+	return this.say;
+};
+
+Inhobitant.prototype.getFriend = function() {
+	return this.hasOwnProperty('friends')
+		? `I have friends: ${this.friends.join(', ')}.`
+		: `I have no friends`;
+};
+
+Inhobitant.prototype.getHandLegs = function() {
+	let legs = `I have ${this.legs} legs`;
+	return this.hasOwnProperty('hands')
+		? `${legs} and ${this.hands} hands`
+		: `${legs}`;
+};
+Inhobitant.prototype.getBio = function() {
```

Comment:
Based on comment above this method shouldn't use `.getHandLegs` as `.getHandLegs` won't be a part of this class anymore.

So, the question is what to do when we need to report hands or legs from child classes?

The answer is method overloading when a child class re-implements parent method, optionally calling that parent method and supplementing it with additional information. Method overloading is an important OOP concept to master.

Reply:
I moved `getHandLeds` to child classes without overloading, should I move it back and use OVERLOADING?

Reply:
should we move on to iteration 2 of code review?